app/controllers/concerns/foreman/controller/smart_proxy_auth.rb

```diff
@@ -32,12 +32,22 @@ def require_smart_proxy_or_login(features = nil)
       return true
     end
 
+    # if we reach this, the requestor is a user and not a smart proxy
+    require_user_login
+  end
+
+  def require_user_login
     require_login
+    check_user_enabled
     unless User.current
```

Comment:
We can return early if one of the first filters fails.
```suggestion
unless require_login && check_user_enabled && User.current
```